pyaerocom/aeroval/setupclasses.py

```diff
@@ -223,112 +220,215 @@ def _get_all_period_strings(self):
-        self.exp_info = ExperimentInfo(exp_id=exp_id)
+    # Pydantic ConfigDict
+    model_config = ConfigDict(arbitrary_types_allowed=True, extra="allow", protected_namespaces=())
 
-        self.time_cfg = TimeSetup()
+    IGNORE_JSON: list[str] = ["_aux_funs"]
+    ADD_GLOB: list[str] = ["io_aux_file"]
 
-        self.modelmaps_opts = ModelMapsSetup()
-        self.colocation_opts = ColocationSetup(
-            save_coldata=True, keep_data=False, resample_how="mean"
-        )
-        self.statistics_opts = StatisticsSetup(weighted_stats=True, annual_stats_constrained=False)
-        self.webdisp_opts = WebDisplaySetup()
+    io_aux_file: Annotated[
+        Path | str, ".py file containing additional read methods for modeldata"
+    ] = ""
 
-        self.processing_opts = EvalRunOptions()
+    _aux_funs: dict = {}
 
-        self.obs_cfg = ObsCollection()
-        self.model_cfg = ModelCollection()
+    proj_id: str
+    exp_id: str
 
-        self.var_web_info = {}
-        self.path_manager = OutputPaths(self.proj_id, self.exp_id)
-        self.update(**kwargs)
+    @computed_field
+    @property
+    def proj_info(self) -> ProjectInfo:
+        return ProjectInfo(proj_id=self.proj_id)
 
+    @computed_field
     @property
-    def proj_id(self) -> str:
-        """
-        str: proj ID (wrapper to :attr:`proj_info.proj_id`)
-        """
-        return self.proj_info.proj_id
+    def exp_info(self) -> ExperimentInfo:
+        return ExperimentInfo(exp_id=self.exp_id)
 
+    @computed_field
     @property
-    def exp_id(self) -> str:
-        """
-        str: experiment ID (wrapper to :attr:`exp_info.exp_id`)
-        """
-        return self.exp_info.exp_id
+    def path_manager(self) -> OutputPaths:
+        if hasattr(self, "model_extra") & bool(
+            cfg_extra_keys := set(self.model_extra).intersection(set(OutputPaths.model_fields))
+        ):
+            subset_dict = {k: self.model_extra[k] for k in cfg_extra_keys}
+            return OutputPaths(proj_id=self.proj_id, exp_id=self.exp_id, **subset_dict)
+        else:
+            return OutputPaths(proj_id=self.proj_id, exp_id=self.exp_id)
+
+    # This is an attempt at a hack to get keys from a general CFG into their appropriate respective classes
+    # It's hard to come up with a way to do this that doesn't introduce breaking changes
+    # Introducing breaking changes only seems appropriate after a config format is settled.
+    # Note: all these computed fields could be more easily defined if the config were
+    # rigid enough to have they explicitly defined (e.g., in a TOML file), rather than dumping everything
+    # into one large config dict and then dishing out get the relevant parts to each class.
+    @computed_field
+    @property
+    def time_cfg(self) -> TimeSetup:
+        if hasattr(self, "model_extra") & bool(
+            time_cfg_keys := set(self.model_extra).intersection(set(TimeSetup.model_fields))
```

Comment:
Is `time_cfg_keys` mean to be the intersection of `set(self.model_extra)` and `set(TimeSetup.model_fields)`?

If so, maybe this cam be refactor:
- without the `intersection` method as `time_cfg_keys := set(self.model_extra) & set(TimeSetup.model_fields)`; or
- without the 2nd set as `time_cfg_keys := set(self.model_extra).intersection(TimeSetup.model_fields)`

Reply:
also, the `&` between the booleans is better written as `and`, like this
`if hasattr(self, "model_extra") and bool(...)`

Reply:
Agreed this looks better.